tracer.go

```diff
@@ -14,13 +16,36 @@ type Tracer interface {
 	//
 	// Examples:
 	//
-	//     var tracer Tracer = ...
+	//     var tracer opentracing.Tracer = ...
 	//
-	//     sp := tracer.StartTrace("GetFeed")
+	//     sp := tracer.StartSpan("GetFeed")
 	//
-	//     sp := tracer.StartTrace("HandleHTTPRequest").
-	//         SetTag("user_agent", req.UserAgent).
-	//         SetTag("lucky_number", 42)
+	//     sp := tracer.StartSpanWithOptions(opentracing.SpanOptions{
+	//         OperationName: "LoggedHTTPRequest",
+	//         Tags: opentracing.Tags{"user_agent", loggedReq.UserAgent},
+	//         StartTime: loggedReq.Timestamp,
+	//     })
 	//
-	StartTrace(operationName string) Span
+	StartSpan(operationName string) Span
+	StartSpanWithOptions(opts *SpanOptions) Span
+}
+
+// StartSpanOptions allows Tracer.StartSpanWithOptions callers to override the
+// start timestamp, specify a parent Span, and make sure that Tags are
+// available at Span initialization time.
+type StartSpanOptions struct {
+	// OperationName may be empty (and set later via Span.SetOperationName)
+	OperationName string
+
+	// Parent may specify Span instance that caused the new (child) Span to be
+	// created. May be nil.
+	Parent Span
+
+	// StartTime overrides the Span's start time, or implicitly becomes
+	// time.Now() if StartTime.IsZero().
+	StartTime time.Time
+
+	// Zero or more Tags. The restrictions on map values are identical to those
+	// for Span.SetTag(). May be nil.
+	Tags map[string]interface{}
```

Comment:
not really zipkin, more like a case of any tracer that uses upfront sampling. If it decides not to sample the trace and marks the span as such, it might also short-circuit saving other init data to the span, even the timestamp, since it assumes the span will never be reported. But if we pass ~~debug~~ `force_sampled=true` tag to the constructor method, tracer won't have any data loss.